tools/Utilities/StatusCheckLogger.cs

```diff
@@ -0,0 +1,146 @@
+﻿using Octokit;
```

Comment:
This file is the primary new functionality. It builds that annotation collection and provides consistent output for error, warning, and informational messages. It also uploads the annotations to GitHub when running in an actions context.